python/az/aro/azext_aro/_aad.py

```diff
@@ -54,6 +56,17 @@ def get_service_principal(self, app_id):
         return None
 
     def create_service_principal(self, app_id):
-        return self.client.service_principals.create(ServicePrincipalCreateParameters(
-            app_id=app_id,
-        ))
+        service_principal = None
+        retries = 0
+        while True:
+            try:
+                service_principal = self.client.service_principals.create(
+                    ServicePrincipalCreateParameters(app_id=app_id))
+                break
+            except GraphErrorException as ex:
+                if retries >= 3:
+                    raise ex
```

Comment:
I think this should be 
```python
if retries >= 3:
    raise
```
from python docs "If no expressions are present, raise re-raises the last exception that was active in the current scope. "

Reply:
That is cleaner. Fixed.